docs/sources/writing-guide/shortcodes/index.md

```diff
@@ -143,6 +143,52 @@ This shortcode inserts a lists of links to the page's subpages and includes the
 {{</* section withDescriptions="true"*/>}}
 ```
 
+## `docs/reference` shortcode
+
+The `docs/reference` shortcode offers more flexible linking than the Hugo builtin `relref` shortcode.
+If content is sourced from one location, and published in multiple locations, the `docs/reference` shortcode helps you control the appropriate link location.
+
+For example, alerting documentation is sourced from the Grafana repository and published to both Grafana documentation and Grafana Cloud documentation.
+Any link from alerting documentation to dashboard documentation, which is also published to both Grafana and Grafana Cloud, needs to be specific as to which destination is appropriate for the audience.
+
+Grafana Cloud alerting documentation should link to Grafana Cloud dashboard documentation, and Grafana alerting documentation should link to Grafana dashboard documentation.
+
+The destination used in a rendered page derives from matching path prefixes:
+
+```markdown
+{{%/* docs/reference */%}}
+[<LABEL>]: "<PROJECT PATH PREFIX> -> <REFERENCE>"
+...
+{{%/* /docs/reference */%}}
+```
+
+- _`LABEL`_ matches the link label used in a reference-style link (`[LINK TEXT][LABEL])`).
+  For more information about reference-style links, refer to [Daring Fireball: Markdown Syntax Documentation](https://daringfireball.net/projects/markdown/syntax#link).
+- _`PROJECT PATH PREFIX`_ is matched against the page’s URL path, where the first matching `PROJECT PATH PREFIX` is chosen.
+  The project of the destinations where you want _`REFERENCE`_ to be used.
+  For example, for Grafana, use `/docs/grafana/`, for Mimir, use `/docs/mimir/`.
+- _`REFERENCE`_ is the parameter that would typically be used in a `relref` shortcode.
+  It can include the syntax `<SOMETHING VERSION>`, which is first looked up in the front matter before falling back to the version inferred from the page's version.
+  Any of the kinds of arguments documented in [Links and Cross References](https://gohugo.io/content-management/cross-references/#use-of-ref-and-relref) can be used.
```

Comment:
I don't understand how this is what we typically use in a relref code and not a partial URL:
/docs/grafana/<GRAFANA VERSION>/dashboards

Reply:
Perhaps we should remove "typically", it's not helpful as our typical usage is probably to use relative paths to files rather than absolute-ish paths like `/docs/grafana/<GRAFANA VERSION>/dashboards`.

Reply:
I think invoking the language of relrefs here is going to be confusing for those of us who still need to look at how relrefs are formatted regularly to just make sure they work at all. Is there another word we can use instead?

Reply:
I don't know if I can because the destination is passed to the relref function.  I have linked to the definitive documentation for relrefs so that people can go there for more information.

I understand the Hugo docs aren't great but I feel like explaining relrefs arguments is outside of the scope of this PR.